exec/java-exec/src/main/java/org/apache/drill/exec/client/DrillClient.java

```diff
@@ -312,6 +312,11 @@ public synchronized void connect(String connect, Properties props) throws RpcExc
     if (connected) {
       return;
     }
+
+    if (props == null) {
```

Comment:
My recommendation is to change other 2 overloaded methods to pass `new Properties()` instead of `null` and making it explicit that `null` is not allowed (avoid passing `null` and checking for `null` at the same time).